### ISSUE TITLE
minimum log level for user dialogs

### ISSUE THREAD

**Issue by jjYBdx4IL** [wxWidgets/wxWidgets#22163]
In order to make the wxWidgets logging system more versatile, it'd be nice to be able to set a minimum log level for dialogs popping up and interfering with user activity. Ie. I want to be able to do a wxLogInfo() without annoying the user and just logging messages to a log widget or a log file. Currently, I just disable the default log implementation completely, which also means that I won't get dialog messages for wxLogError.

**Comment by vadz** [comment]
You already have the possibility to do this with `wxLog::SetLogLevel()` and even `SetComponentLevel()`, without even speaking of defining your custom log target which can handle different levels as flexibly as you wish.

**Comment by jjYBdx4IL** [comment]
SetLogLevel has no effect on those modal dialog popups. Only thing I can do is to replace the the active log target at the start.

**Comment by vadz** [comment]
Strange, could you please provide a simple example reproducing the problem?

**Comment by jjYBdx4IL** [comment]
let me rephrase it: SetLogLevel is global. It doesn't allow me to tell a logger to only log messages above a certain level. At least that's my understanding and experience using it.

**Comment by vadz** [comment]
Yes, `SetLogLevel()` is global but if you want to ignore the messages with priority less than the given one, this is exactly what you want, don't you?

Maybe you should define what exactly do you want to achieve first before we try to find the solution.

**Comment by jjYBdx4IL** [comment]
"minimum log level for user dialogs" to pop up

not sure what's to misunderstand there :)

I think it's quite obvious that you don't want to annoy users with *verbose* messages, for example.

**Comment by vadz** [comment]
> not sure what's to misunderstand there :)

Please provide a patch to the minimal sample showing your point.
 
> I think it's quite obvious that you don't want to annoy users with _verbose_ messages, for example.

Calling `wxLog::SetLogLevel(wxLOG_Status)` should do it.
